app/views/session.py

```diff
@@ -35,15 +36,17 @@ def login():
     if session:
         session.clear()
 
-    logger.debug("attempting token authentication")
-    authenticator.jwt_login(request)
-    logger.debug("token authenticated - linking to session")
+    token = decrypt_token(request.args.get('token'))
+    jti_token = token.get('jti')
+    if jti_token is not None and not use_token(jti_token):
```

Comment:
Maybe allow use_token to take None and return False if so to simplify this and make use_token more resilient? Should we also check the content of the jti is correct/valid (the right size/length or format or similar?)